### PR TITLE
Move xctestrun creation into new template script

### PR DESCRIPTION
**Pull request by mattrobmattrob**
Refactor to prepare for using the xctestrun creation script outside of the test runner.

Work towards https://github.com/bazelbuild/rules_apple/issues/2175.

# Testing

```
$ bazel test --local_test_jobs=1 //test:ios_xctestrun_runner_ui_test //test:ios_xctestrun_runner_unit_test
INFO: Build option --test_env has changed, discarding analysis cache.
INFO: Analyzed 2 targets (0 packages loaded, 1048 targets configured).
INFO: Found 2 test targets...
INFO: Elapsed time: 336.777s, Critical Path: 336.45s
INFO: 3 processes: 1 internal, 2 darwin-sandbox.
INFO: Build completed successfully, 3 total actions
//test:ios_xctestrun_runner_ui_test                                      PASSED in 102.8s
//test:ios_xctestrun_runner_unit_test                                    PASSED in 233.0s

Executed 2 out of 2 tests: 2 tests pass.
```

### CODE REVIEW

File: apple/testing/default_runner/ios_xctestrun_runner.template.sh [hunk 1]
```diff
@@ -384,32 +188,29 @@ fi
...
+  xctestrun_command_line_args=""
+  if [[ -n "${command_line_args:-}" ]]; then
+    xctestrun_command_line_args="${command_line_args[@]}"
+  fi
 
+  readonly xctestrun_file="$test_tmp_dir/tests.xctestrun"
+  test_env="$test_env" \
+    TEST_FILTER="$TEST_FILTER" \
+    command_line_args="$xctestrun_command_line_args" \
+    ./%(xctestrun_creator.sh)s \
```
Comment (mattrobmattrob) [inline apple/testing/default_runner/ios_xctestrun_runner.template.sh:204]: I don't love this arg and env var passing mechanism all that much. Let me know if you've better ideas. I figure we can make it more explicit in follow up work as we add usages of the xctestrun creation script.

**Comment by luispadron** [comment]
@mattrobmattrob Was this ready for review, i can take a look if so!

**Comment by mattrobmattrob** [comment]
Should be, @luispadron!

**Comment by aaronsky** [comment]
@mattrobmattrob any reason you closed this besides time? I can try and pick this up if you think it's still needed. We're going to need the ultimate functionality of #2175 at some point anyway. 🙂

**Pull request closed without merging**